### ISSUE TITLE
OpenAuto crashed with RPi 3A+

### ISSUE THREAD

**Issue by Compizfox** [htruong/crankshaft#447]
- What phone do you have?
N/A

- What crankshaft version?
Alpha7.2 2021-02-09

- Steps to reproduce the bug, if you're planning to report a bug. Please indicate whether the bug is always repoducible.
Flash the image to a SD card, and boot it. Yes.

After boot, I get the following error:

```
Unfortunately, OpenAuto crashed.
Please report this indicident to Crankshaft
Restart in 5 seconds...
```

I'm using a HDMI screen.

I tried both EGL and X11. I have the same issue with the alpha7 2020-10-26 build.

I'm not quite sure how to create a debug log, since I don't get far enough to "push the gray button on bottom of main screen". I tried by just booting in debug mode anyway but I don't get any log.

**Comment by matt2005** [comment]
I've not tried a RPi 3A+, but it should work but it will need debugging as it's not a hardware config i've tested.

[Enable Dev_Mode](https://github.com/opencardev/crankshaft/wiki/Dev-Mode-and-Debug-Mode#how-to-enable) 
login to ssh

To create debug file from ssh run
```text
crankshaft debuglog
```

**Comment by matt2005** [comment]
@Compizfox Can you try the instructions above?

**Comment by Compizfox** [comment]
[debug.zip](https://github.com/opencardev/crankshaft/files/6054945/debug.zip)

Here you go.

**Comment by scott9000** [comment]
I am also trying to run crankshaft on rpi 3a+ and getting the same issue. Would love to find out if this gets resolved 🙂

**Comment by matt2005** [comment]
@scott9000 Can you create a debug file? please ensure that you are not in x11 mode.

**Comment by scott9000** [comment]
Yeah will do. Will send it to you when I'm finished work. Thank you!

**Comment by scott9000** [comment]
[debug.zip](https://github.com/opencardev/crankshaft/files/6066226/debug.zip)
let me know if this is what you need. I'm a bit of a noob but can follow instrucions

**Comment by matt2005** [comment]
@scott9000 what is your setup? What attached devices? Screen? Etc

**Comment by scott9000** [comment]
I have a 5" elecrow LCD touch screen and pixel 4, the screen is working
fine but no GUI

**Comment by Compizfox** [comment]
> @scott9000 Can you create a debug file? please ensure that you are not in x11 mode.

The debug log I submitted was while running in X11 mode. Do you need one in EGL mode from me as well?

**Comment by matt2005** [comment]
@Compizfox The logs appear to show crashing. Can you try editing the gpu_mem in /boot/config.txt. Try editing the gpu_mem from 256 to 128. 
```bash
# GPU Mem
gpu_mem=128
```

**Comment by Compizfox** [comment]
Thanks, that did the trick! OpenAuto is now starting correctly.

However, when connecting my phone (over USB) nothing happens. Am I missing something or should I create a new issue?

**Comment by matt2005** [comment]
do you have Androidauto installed on Phone? You might need to connect via a powered usb hub, as the pi 3a might not have enough power over usb.
Alternatively try wireless mode see wiki for details

**Comment by Compizfox** [comment]
> do you have Androidauto installed on Phone?

Yes, I have Android Auto installed and when I connect it over USB, it says "connected to car" on the phone. However, nothing happens on the OpenAuto head unit.

> You might need to connect via a powered usb hub, as the pi 3a might not have enough power over usb.

Hmm, I'm afraid I don't have one on hand. Why does it need to power the phone anyway? I'm not expecting it to charge it or anything, lol

>  Alternatively try wireless mode see wiki for details

Which wiki page exactly? My first idea was to try Bluetooth, but I couldn't find out how to pair it. The [Bluetooth wiki page](https://github.com/opencardev/crankshaft/wiki/Bluetooth) says to "tap the `BT-Pairing` button" but I don't see that...

**Comment by matt2005** [comment]
[AA wireless](https://github.com/opencardev/crankshaft/wiki/Android-Auto-Wifi)
 You will need to enable the onboard BT, I've updated the wiki page for BT.

Enable onboard BT. You must reboot after this.
```bash
crankshaft bluetooth builtin
sudo reboot
```

**Comment by matt2005** [comment]
Did the above work for you? @Compizfox

**Comment by Compizfox** [comment]
@matt2005 No, unfortunately not.

I was able to enable Bluetooth and pair my phone, but that appears to be for audio only?

I tried wireless Android Auto mode over WiFi, but that doesn't work either. When I connect my phone to Crankshaft device over WiFi it seems to detect it (the top-left button gains a Android Auto icon, but when I press on it nothing happens.

EDIT: Sorry, I was too quick. I managed to get it working (over USB) with a different power supply. So it appears to be a power issue after all, weird...
